src/server/src/main/java/io/cassandrareaper/service/StreamManager.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.service;
+
+import io.cassandrareaper.AppContext;
+import io.cassandrareaper.ReaperException;
+import io.cassandrareaper.core.Cluster;
+import io.cassandrareaper.core.Node;
+import io.cassandrareaper.core.Stream;
+import io.cassandrareaper.core.StreamSession;
+import io.cassandrareaper.jmx.JmxProxy;
+import io.cassandrareaper.jmx.StreamStatusHandler;
+import io.cassandrareaper.storage.StreamCache;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import javax.management.openmbean.CompositeData;
+
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Futures;
+import org.apache.cassandra.streaming.ProgressInfo;
+import org.apache.cassandra.streaming.SessionInfo;
+import org.apache.cassandra.streaming.management.ProgressInfoCompositeData;
+import org.apache.cassandra.streaming.management.SessionInfoCompositeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements the functionality needed for receiving,
+ * handling and serving streaming-related notifications.
+ */
+public final class StreamManager implements StreamStatusHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StreamManager.class);
+  private final AppContext context;
+  private final ExecutorService executor = Executors.newFixedThreadPool(5);
+  private final StreamCache streamSessions = new StreamCache();
```

Comment:
so if i understand this correct…

the streaming overview presented by reaper is solely in-memory, and limited to the past 24 hours.
this means that different reaper instances can present different views of streaming plans.

it's great that an incremental solution is being rolled out, but still i think this deserves some design docs, and to be clearly marked `@Alpha` of its new classes.